src/main/java/Task.java

```diff
@@ -0,0 +1,27 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "[X] " + this.description : "[] " + this.description); // mark done task with X
+    }
+
+    public void setDone() {
+        this.isDone = true;
+    }
+
+    public void setNotDone() {
+        this.isDone = false;
+    }
+
+    public String toRecord() {
+        return (isDone ? "1|" : "0|") + description;
```

Comment:
Ooh, this would like really nice in the save file! Should `Duke#takeRequest()` somehow handle the `"|"` character if it's given in a `userInput`?